projects/common/src/time/page-time-range.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { isNil } from 'lodash-es';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import { distinctUntilChanged, map, share, takeUntil } from 'rxjs/operators';
+import { PreferenceService, StorageType } from '../preference/preference.service';
+import { TimeRange } from './time-range';
+import { TimeRangeService } from './time-range.service';
+
+@Injectable({ providedIn: 'root' })
+export class PageTimeRangeService implements OnDestroy {
```

Comment:
Why did you not combine it with the existing TR service?

Reply:
You mentioned the time range service should not be dependent on the preference service ideally, and because the logic within `PageTimeRangeService` is specific to being used with `PageTimeRangeComponent` and `NavItemComponent` and not all time ranges.

Reply:
Maybe though, the `PageTimeRangeService` should live in the same directory as `PageTimeRangeComponent`, under `projects/components/src/time-range/page-time-range` instead ?.  I think when i originally created the files i had slightly different intentions for their use.